src/Config.ts

```diff
@@ -659,9 +663,7 @@ export class Config implements ConfigParams, ParserConfig {
     validateNumberToBeAtLeast(this.precisionEpsilon, 'precisionEpsilon', 0)
     this.useColumnIndex = configValueFromParam(useColumnIndex, 'boolean', 'useColumnIndex')
     this.useStats = configValueFromParam(useStats, 'boolean', 'useStats')
-    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-    // @ts-ignore
-    this.binarySearchThreshold = undefined
+    this.binarySearchThreshold = binarySearchThreshold ?? Config.defaultConfig.binarySearchThreshold
```

Comment:
Unless I am terribly mistaken, this PR missed the most important point, that `binarySearchThreshold` is not used anywhere. 

Therefore, it should be removed as a member of this class and from the default config object.

We can keep the warning if someone tries using it.

Reply:
As a continuation of the original PR (https://github.com/handsontable/hyperformula/pull/830) I tried to focus on fixing the double warnings and covering the change with tests only. To keep the PR as simple as possible with the possibility to release the patch version.

Removing the `binarySearchThreshold` option from defaults would generate a new TS definition. I don't know if that wouldn't be a breaking change?

Reply:
Yes, you are right that it would be an unnecessary breaking change. With your change, it is possible to release this as a PATCH version.